e2e/test/iothub/DeviceStreamingE2ETests.cs

```diff
@@ -0,0 +1,392 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Client.Exceptions;
+using Microsoft.Azure.Devices.E2ETests.Helpers;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using ClientDeviceStreamingRequest = Microsoft.Azure.Devices.Client.DeviceStreamRequest;
```

Comment:
I don't love that we have the same type name in the device and service clients. Sure, normally someone won't be using both together in an application, but there are cases, especially in tests and samples.

I'd propose we name them differently, so they don't conflict. Thoughts on names?

Perhaps on the service it can be `DeviceStreamingRequest/Response` but on the device it'd just be `StreamingRequest/Response`.

Reply:
"DeviceClientStreamingRequest" vs "ServiceClientStreamingRequest"
"DeviceToServiceStreamingRequest" vs "ServiceToDeviceStreamingRequest"

These extra distinctions feel a bit clunkier then just using the same names in the device and service client

Reply:
My preference:
1. `DeviceToServiceStreamingRequest` and `ServiceToDeviceStreamingRequest`
2. `Devices.Client.DeviceStreamRequest` and `Devices.DeviceStreamRequest` (or alias -> what we have now)
3. `DeviceClientStreamingRequest` and `ServiceClientStreamingRequest` (I don't particularly like putting the term "client" in there)

Also, a note on the existing names - we call it "device" streaming, but I am seeing that we support it on the module client as well :(

Should we call it `ClientToServiceStreamingRequest`/`Response`?

Reply:
I'll just leave it as is for now.